### PR TITLE
Add menu-dropdown for Donate

### PR DESCRIPTION
**Pull request by FelicityZhang**

### CODE REVIEW

File: src/components/MerchantsPage/MerchantCard.tsx
Comment (justintmckibben) [inline src/components/MerchantsPage/MerchantCard.tsx:91]: nit: Mind removing this whitespace?

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:26]: nit: DropdownOpen should be lower camel-cased

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:91]: nit: remove

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:97]: You should use ReactRouter to reroute here vs an href

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:98]: Same here

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:145]: Anything with `merchant.*` can be routed via a ReactRouter vs redirecting and reloading the entire page. If we redirect like this, we are forgoing the real benefit of using React, which is to only reload the elements in the DOM that have changed. If you reload our SPA with a redirect, it loads everything from scratch

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:172]: nit: remove extra whitespaces

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:209]: nit: add space after colon

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:220]: nit: no need for this whitespace

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:251]: same here

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:322]: nit: the newlines and whitespaces here are slightly off

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:333]: What's the z index for?

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:339]: I know that we don't do this in our codebase, but we really should. It's general accepted convention to keep margin widths, etc. even numbers. This should be reflected in designs too.

cc @nanxiy what do you think about establishing a "multiple of 8px" rule? or at the very least a multiple of 2px

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:352]: In general, negative pxs are a code smell. We try to avoid them unless on a tight deadline

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:341]: Do we need this?

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:335]: There is a lot going on here. Why do you need display block, flex and position absolute?

**Review by justintmckibben — COMMENTED** [review]
Looks good so far! I have some open questions. Also you can use yarn format and/or yarn eslint --fix to address CSS/JS/whitespace formatting nits all at once

File: src/components/Navbar/NavBar.tsx
Comment (aliceluo2048) [inline src/components/Navbar/NavBar.tsx:89]: Why was this changed from /merchants?

**Comment by aliceluo2048** [comment]
Can you also include a screenschot/gif of the dropdown?

File: src/components/Navbar/NavBar.tsx
Comment (nanxiy) [reply src/components/Navbar/NavBar.tsx:339]: Yes we definitely can do that for our unique designs but for Weebly.. I don't think I have control over that. I would reference our live page for their specs

Comment (justintmckibben) [reply src/components/Navbar/NavBar.tsx:339]: Sounds good! Square follows a standard for multiples of 8px which is why I suggested it, or 2px if you need spacing < 8px. I know some companies that do multiples of 5px — so whatever you think would look good

Comment (justintmckibben) [reply src/components/Navbar/NavBar.tsx:339]: I think Weebly we can just keep it as is

**Comment by FelicityZhang** [comment]
Screen Shot 2020-09-14 at 10 17 03 AM
Screen Shot 2020-09-14 at 10 18 46 AM

File: src/components/Navbar/NavBar.tsx
Comment (FelicityZhang) [reply src/components/Navbar/NavBar.tsx:333]: Without the z index, the DescriptionBox would be on top of the dropdown container on the merchant page.

File: src/components/Navbar/NavBar.tsx
Comment (aliceluo2048) [inline src/components/Navbar/NavBar.tsx:138]: Fyi on the weebly nav bar the dropdown appears when we hover

File: src/components/MerchantsPage/MerchantCard.tsx
Comment (justintmckibben) [reply src/components/MerchantsPage/MerchantCard.tsx:91]: Bump!

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:102]: Could you make this one to=... as well?

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:160]: to=...

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [reply src/components/Navbar/NavBar.tsx:241]: bump

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [reply src/components/Navbar/NavBar.tsx:350]: I see. Mind making a constant called $max-z-index or something and making it 9999 or something, and importing that?

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:351]: Should be an even number

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:354]: all of these should be multiples of 2

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:390]: Does this color not exist as a variable anywhere?

File: src/components/Navbar/NavBar.tsx
Comment (justintmckibben) [inline src/components/Navbar/NavBar.tsx:394]: Can you run yarn eslint --fix to fix these whitespace issues? There should be a space after the :

**Review by justintmckibben — COMMENTED** [review]
Still have a few nits, but it's almost there!

**Review by justintmckibben — APPROVED** [review]
LGTM! Mind getting a frontend engineer's review?

**Comment by justintmckibben** [comment]
@FelicityZhang what's the update on this?

**Comment by FelicityZhang** [comment]
> @FelicityZhang what's the update on this?

I was waiting for the reviews. Seems no reviews so far. I hope I didn't miss anything. : )

**Comment by ath-chen** [comment]
@FelicityZhang Nice job! 

These are just suggestions, so feel free to use them or not. 

One way you can reduce code for the dropdown & mobile dropdown menus is by storing all the info in an object, then just mapping over it once. This will be esp helpful if we continue to add additional links to this dropdown. Something like this, but feel free to make it however you like. 

Another thought, should we store the dropdown mapping as a separate function, then call it on lines 99 & 155 so we can avoid writing it twice? And we could use media queries to display diff css when its mobile or desktop

```
const dropdownOptions = [
  { url: "/merchants", translation: "donate", external: false },
  { url: "/gift-a-meal-home", translation: "gift-a-meal", external: false },
  { url: "https://stickylocals.com/scl", translation: "sticky-locals-x-scl", external: false  },
  { url: "https://www.friendofafriend.studio/shop", translation: "friendofafriend-studioxscl", external: false }
]

dropdownOptions.map(option => {
  const { url, translation, external} = option;
  return (
    {
       external ? (
            <MobileItem href={url}>
               { t(`navBar.header.waystodonate.${translation}`) }
            </MobileItem>
        ) : (
           <MobileItemLink to={url}>
             { t(`navBar.header.waystodonate.${translation}`) }
          </MobileItemLink>
        )
    }
  )
})
```

**Review by ath-chen — APPROVED** [review]

**Comment by FelicityZhang** [comment]
> @FelicityZhang Nice job! Approved the PR!
> 
> These are just suggestions, so feel free to use them or not.
> 
> One way you can reduce code for the dropdown & mobile dropdown menus is by storing all the info in an object, then just mapping over it once. This will be esp helpful if we continue to add additional links to this dropdown. Something like what I did below.
> 
> Another thought, should we store the dropdown mapping as a separate function, then call it on lines 99 & 155 so we can avoid writing it twice? And we could use media queries to display diff css for mobile / desktop views
> 
> ```
> const dropdownOptions = [
>   { url: "/merchants", translation: "donate", external: false },
>   { url: "/gift-a-meal-home", translation: "gift-a-meal", external: false },
>   { url: "https://stickylocals.com/scl", translation: "sticky-locals-x-scl", external: true  },
>   { url: "https://www.friendofafriend.studio/shop", translation: "friendofafriend-studioxscl", external: true }
> ]
> 
> dropdownOptions.map(option => {
>   const { url, translation, external } = option;
>   return (
>     {
>        external ? (
>             <MobileItem href={url}>
>                { t(`navBar.header.waystodonate.${translation}`) }
>             </MobileItem>
>         ) : (
>            <MobileItemLink to={url}>
>              { t(`navBar.header.waystodonate.${translation}`) }
>           </MobileItemLink>
>         )
>     }
>   )
> })
> ```

**Comment by FelicityZhang** [comment]
@ath-chen Thank you so much! I love the idea.  I just made the changes. @justintmckibben

File: src/components/Navbar/NavBar.tsx
Comment (aliceluo2048) [inline src/components/Navbar/NavBar.tsx:8]: What is this?

**Review by jacoblb64 — APPROVED** [review]
Looks good to me--feel free to resolve any of the comments you've addressed to cleanup how noisy it is on here.

File: src/components/Navbar/NavBar.tsx
Comment (FelicityZhang) [reply src/components/Navbar/NavBar.tsx:8]: I'm not sure if I added it. should I delete it ?

File: src/components/MerchantsPage/gam/GiftAMealPage.tsx
Comment (aliceluo2048) [inline src/components/MerchantsPage/gam/GiftAMealPage.tsx:85]: This is causing a build error and was incorrectly included in the resolving the merge conflicts

File: src/components/Navbar/NavBar.tsx
Comment (aliceluo2048) [reply src/components/Navbar/NavBar.tsx:8]: Maybe an accident import. It's not used anywhere so you can remove it.

File: src/components/Navbar/NavBar.tsx
Comment (aliceluo2048) [inline src/components/Navbar/NavBar.tsx:54]: nit: You don't really need to wrap this in a function, assigning it to a variable is fine. Also the variable name should be camelCase.

Anyway feel free to fix this after you merge your PR.

File: src/components/Navbar/NavBar.tsx
Comment (aliceluo2048) [inline src/components/Navbar/NavBar.tsx:68]: nit: You don't really need to wrap this in a function, assigning it to a variable is fine. Also the variable name should be camelCase.

Not a blocking issue - go ahead merge the PR. 

Unrelated but for clean code, you can just do this
`const drop = () =>
    dropdownOptions.map(option => {
... })
`

**Pull request merged**
